drivers/temperature/adt7420/adt7420.c

```diff
@@ -419,17 +411,36 @@ int adt7420_spi_reg_read(struct adt7420_dev *dev,
  * @return register_value  - Value of the register.
 *******************************************************************************/
 int adt7420_i2c_reg_read(struct adt7420_dev *dev,
-			 uint8_t register_address, uint8_t *data)
+			 uint8_t register_address, uint16_t *data)
 {
-	uint8_t register_value = 0;
+	uint8_t data_buffer[3] = { 0, 0 };
+	uint8_t num_bytes;
+
+	switch (register_address) {
+	case ADT7420_REG_STATUS:
+	case ADT7420_REG_CONFIG:
+	case ADT7420_REG_ID:
+	case ADT7420_REG_HIST:
+	case ADT7420_REG_T_HIGH_LSB:
+	case ADT7420_REG_T_LOW_LSB:
+	case ADT7420_REG_T_CRIT_LSB:
+		num_bytes = 1;
+		break;
+	default:
+		num_bytes = 2;
+		break;
+	}
 
 	if (no_os_i2c_write(dev->i2c_desc, &register_address, 1,
 			    0)) //add a repeat start
 		return -1;
-	if (no_os_i2c_read(dev->i2c_desc, &register_value, 1, 1))
+	if (no_os_i2c_read(dev->i2c_desc, data_buffer, num_bytes, 1))
 		return -1;
 
-	*data = register_value;
+	if (num_bytes == 1)
+		*data = data_buffer[0];
+	else
+		*data = data_buffer[1] | (data_buffer[0] << 8);
```

Comment:
use `no_os_get_unaligned_be16`

Reply:
Utilized the no_os_get_unaligned_be16 for this